server/app/database_models.py

```
@@ -0,0 +1,342 @@
+        self.db_name = database_name
+        self._create_table()
+        
+
+    def _create_table(self) -> None:
+        """
+        Private method. Creates the project table if it does not yet exist, otherwise does nothing.
+        """
+        try:
+            with sqlite3.connect(self.db_name) as conn:
+                cur = conn.cursor()
+                cur.execute("PRAGMA foreign_keys = ON;")
+
+                cur.execute(f"""
+                    CREATE TABLE IF NOT EXISTS {PROJECT_TABLE_NAME} (
+                        project_id INTEGER PRIMARY KEY AUTOINCREMENT,
+                        name TEXT NOT NULL UNIQUE,
+                        description TEXT NOT NULL,
+                        created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+                    )
+                """)
+
+        except sqlite3.Error as e:
+            print(f"There was an error creating the table: {e}")
+
+    def insert(self, project_name: str, description: str = "") -> Optional[int]:
+        """
+        Method to insert propject into the table. Returns the id of the project on success, and None on failure.
+
+        Args:
+            project_name (str): The name of the project. Must be unique or else it will fail on insert
+            description (str): The description of the project
+
+        Returns:
+            Optional int: The project id on successful insert, or None on failure.
+        """
+        try:
+            project_id = None
+
+            with sqlite3.connect(self.db_name) as conn:
+                cur = conn.cursor()
+                cur.execute("PRAGMA foreign_keys = ON;")
+
+                cur.execute(f"""
+                    INSERT INTO {PROJECT_TABLE_NAME} (name, description) VALUES (?, ?)
+                """, (project_name, description))
+
+                project_id =  cur.lastrowid
+
+            return project_id
+
+        except sqlite3.Error as e:
+            print(f"There was an error inserting into the table: {e}")
```

It might be better to just throw an exception here